### ISSUE TITLE
Failure to parse a Java multiline string ending in an escaped double quote, i.e. four consecutive double quotes 

### ISSUE THREAD

**Issue by begglez** [semgrep/semgrep#9991]
**Describe the bug**
Semgrep can't parse  a Java multiline string ending in an escaped double quote.

**To Reproduce**
Try parsing this:
`String x = """`
`hello world`
`\"""";`

Not linking to the semgrep.dev playground because I didn't see a way to permalink a snippet without creating an account.

**Expected behavior**
Parse the string

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Environment**
Reproducible in semgrep.dev

**Use case**
Eliminate erroneous alerts from hooks on GH PRs

**Comment by ievans** [comment]
thanks for the bug report!

ftr, you can create a permalink by clicking the "share" button without login

**Comment by yosefAlsuhaibani** [comment]
Hi @begglez: I was going through our backlog and saw this issue; would something like https://semgrep.dev/playground/s/8K89J be expected?

**Comment by begglez** [comment]
Yes, looks good! Unfortunately there's no playground link in my original report but I think it was functionally identical to that. I guess this bug got fixed in the last 18 months. Closing
